### PR TITLE
[WIP][8.0][BR001024] - Added demo data for BR modules

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
- Added demo data for BR modules

@elicoidal @victormartinelicocorp

### CODE REVIEW

**Comment by serpentcs-dev1** [comment]
@elicoidal @victormartinelicocorp 

Please review it

cc
@sudhir-serpentcs

File: business_requirement/__openerp__.py
Comment (elicoidal) [inline business_requirement/__openerp__.py]: I dont want to add one extra dependency only for demo data.
Please change the Demo Portal user to Admin User

File: business_requirement_deliverable_resource_template/data/business_requirement_deliverable_resource_template.xml
Comment (elicoidal) [inline business_requirement_deliverable_resource_template/data/business_requirement_deliverable_resource_template.xml]: File name

File: business_requirement_deliverable_resource_template/data/business_requirement_deliverable_resource_template.xml
Comment (elicoidal) [inline business_requirement_deliverable_resource_template/data/business_requirement_deliverable_resource_template.xml]: remove dead code

**Comment by elicoidal** [comment]
Tests: 
- [x] Products in deliverables and resources lines seems all wrong (Probably linked to commented code)
- [x] In BR Business Analysis deliverables, please change 800 => 50 and 2000 by 45
- [x] In BR Gap analysis deliverables, please change 800 => 50 and 0 => 35
- [ ] Category for all products should be "All / Saleable / Services"
- [x] I dont know why but I have 2 tabs "Default Resource lines" in Product form
- [ ] WEB02 is missing some resources lines
- [x] For BR and Project category, please change from "CAPS" to "Normal" (SYSTEM => System)

To be tested after #192 is merged as there might be some light changes in the structure.

**Review by elicoidal — CHANGES_REQUESTED** [review]

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal could you please review.please correct me if I am wrong

> Products in deliverables and resources lines seems all wrong (Probably linked to commented code)
- Probably code has improved but needs review

> In BR Business Analysis deliverables, please change 800 => 50 and 2000 by 45
- Changed Values 

> In BR Gap analysis deliverables, please change 800 => 50 and 0 => 35
- Changed Values 

> Category for all products should be "All / Saleable / Services"
- product.product default product category  is `All` so no need to change

>  I dont know why but I have 2 tabs "Default Resource lines" in Product form

- I have seen there is  same name `resource_lines` used but different models 

- one resource_lines for `business.requirement.resource` object and

- second one resource_lines for `business.requirement.resource.template` with the same name

- so i have improved second number one2many  from `resource_lines`  to  `resource_template_lines`

> WEB02 is missing some resources lines

- Improved code
> For BR and Project category, please change from "CAPS" to "Normal" (SYSTEM => System)

- Improved code in `business.requirement.category` and  `project.category.main` for demo data

**Comment by elicoidal** [comment]
@darshan-serpent @YogeshMahera-SerpentCS 
Still some glitches:
- [ ] First 2 deliverables are not linked to products ([BR000001] Corporate Website and [BR000003] ERP Setup)
- [ ] in product there should be only one tab "Resource template lines" (I dont know what is the second one)
- [ ] change "Website product" name to "Website Template"
- [ ] in task category change Pm to PM
- [ ] I cannot see any RL template in the products
- [ ] Change the product category as suggested
- [ ] (not sure this is for this PR): The cost control tab is wrong (no total for tasks)

**Comment by YogeshMahera-SerpentCS** [comment]
>First 2 deliverables are not linked to products ([BR000001] Corporate Website and [BR000003] ERP Setup)
@elicoidal which product should be linked ?

- Developer
- Designer
- Consultant
- Website Template 
- Website Design
- Odoo Quick Start Package

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS please refer to the original documentation (mail):
A.
Description= Corporate Website
Responsible = Admin
Priority =2
Category=WEBSITE, DESIGN
Reviewer=Demo
To be Reviewed= True
Stage=Confirmed
Customer= Agrolait
Deliverable: use Website Design product (Quantity = 1, Sales Price=5200)
Product=[E-COM09] iMac, quantity = 1 unit, type = Procurement, name=Server
Product=[WEB01] Website Template, Quantiy=1 unit, type=Procurement, name=Beautiful template
product=Designer (Name = Customer design assistance, 3 days, Task, assigned to Demo Portal, Task category= DESIGN, cost=35, unit sales price=60)
Product=Developer (Name = Interface Development, 7 days, Task, assigned to Demo, Task category= DEVELOPMENT, cost=25, unit sales price=35)

C.
Description= Server Set up
Responsible = Demo
Priority =2
Category=ERP, INFRA
Reviewer=Demo Portal
To be Reviewed= False
Parent BR = ERP Implementation
Stage=Stakeholder
Customer= Yourcompany
Deliverable: Product= Developer, Name=Server Setup, 8h
Product=Developer (Name= Server Setup, 8h, Task, assigned to Demo, Task category= DEVELOPMENT, cost=25, unit sales price=35)

**Comment by YogeshMahera-SerpentCS** [comment]
>  I cannot see any RL template in the products

@elicoidal 
I don't get this point.

You mean that `Resource template lines`

All Resouce lines which having same products will be linked here right.?

products odoo

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS no. Template lines should contain the default value I have provided you.
This template lines are added to the BR resources when the deliverable is selected.
cc @sudhir-serpentcs 

for example:
 E.
Name= Website Design
Internal Ref=WEB02
Unit=unit
Type=service
Cost =3500
Sales price = 5200
Contains following template Resource lines
Product=[E-COM09] iMac, quantity = 1 unit, type = Procurement, name=Server
Product=[WEB01] Website Template, Quantiy=1 unit, type=Procurement, name=Beautiful template
product=Designer (Name = Customer design assistance, 3 days, Task, assigned to Demo Portal, Task category= DESIGN, cost=35, unit sales price=60)
Product=Developer (Name = Interface Development, 7 days, Task, assigned to Demo, Task category= DEVELOPMENT, cost=25, unit sales price=35)

F.
Name=Odoo Quick Start Package
Internal Ref=OD01
Sales Price = 4000
Cost Price = 1500
Contains following template Resource lines
Product=consultant (Name= Functional Analysis, 24h, Task, assigned to Admin, Task category= ANALYSIS, cost=30, unit sales price=50)
Product=developer (Name=Technical Support, 32h, Task, assigned to Demo, Task category= DEVELOPMENT, cost=25, unit sales price=35)

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Could please review.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS still many problems:

- [ ] Tasks Category are missing
screenshot from 2017-07-17 14-54-48

- [ ] Website Design and Odoo Quick Start Package product are duplicated
- [ ] Deliverable Server Setup price should be 35
- [ ] Many resources lines are missing in BR (eg: Business Analysis)
etc.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS is it testable?

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Still **WIP**
Once complete will update you.

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Could please test it again

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS Almost there!
- [ ] on BR0001:
* A1090: in RL sale price should be 1799, cost 1299
 *Web01: Sales price: 150/Cost: 100
* SER01: Sales price: 280/200
* SER02: Sales Price:480/280
- [ ] in BR003 / Deliverable, Cost control information is wrong (not sure this is a demo data issue though)
screenshot from 2017-07-21 14-29-24

- [ ] SO0008 Unit price should be 50 and 45 (instead of 800 and 2000)

@sudhir-serpentcs @darshan-serpent I think we might need to wait for #192 to be merge, then rebase and probably fix the project link to the BR
cc @victormartinelicocorp can you review?

**Comment by elicoidal** [comment]
Another small additional topic. Can you add the following text when the screen of BR is empty (Please format it as a list for the second part of the message):
screenshot from 2017-07-21 14-56-45

```
         <p class="oe_view_nocontent_create">
            Click to start a new Business Requirement. 
          </p><p>
A Business requirement (BR) is the expression of a business need by a customer or internal project user.
A BR contains multiple different parts to explain the stakeholder need and how to meet his/her requirements:

    Customer Story: this is the requirement as expressed by the customer
    Scenario: How/where the current solution can provide a suitable scenario to answer the customer story
    Gap: For the uncovered part of the scenario, elaborate the gap/need for specific developments/setup
    Test case: A set of conditions under which a tester will determine whether the application, software system or one of its features is working as it was originally established for it to do.
    Deliverables to be provided to the customer/user
    Resources necessary to achieve the deliverables
    Additional information (approval, cost control etc.)

          </p>
```

**Comment by YogeshMahera-SerpentCS** [comment]
>  in BR003 / Deliverable, Cost control information is wrong (not sure this is a demo data issue though)

I can not find any wrong data same

Description= ERP Setup
Responsible = Demo
Priority =1
Category=ERP
Reviewer=Demo Portal
To be Reviewed= False
Parent BR = ERP Implementation
Stage=Approved
Customer= Yourcompany
Deliverable: use Odoo Quick Start (Qty=1, Sales Price=4000)
Add all resources:
Product=consultant (Name= Functional Analysis, 24h, Task, assigned to Admin, Task category= ANALYSIS, cost=30, unit sales price=50)
Product=developer (Name=Technical Support, 32h, Task, assigned to Demo, Task category= DEVELOPMENT, cost=25, unit sales price=35)

dl

@elicoidal

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS 
in above example you provided, in the deliverable header, Total tasks should be 1520, Gross Profit should be 4000-1520 = 3480
screenshot from 2017-07-21 18-23-06
Let me know if unclear

**Comment by YogeshMahera-SerpentCS** [comment]
Total tasks is 1520,
Gross Profit is 4000-1520 = 2480

test1

**Comment by elicoidal** [comment]
Did you fix SO008 too?

**Comment by YogeshMahera-SerpentCS** [comment]
yes fixed SO008

**Comment by elicoidal** [comment]
SO008 is OK but not the Cost control.
By the way, SO008 is validated: it should be in draft state
@YogeshMahera-SerpentCS

**Comment by YogeshMahera-SerpentCS** [comment]
SO008 is on draft stage

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Could you please review again.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS The cost control tab is still wrong.
SO0008 status is OK
in the following screenshot, expected task total = 480 (same would be true for procurement if any in the resources lines)
screenshot from 2017-07-24 18-20-31

cc: @sudhir-serpentcs

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
Yes, just checked and I already made changes in py files.

test3

**Review by elicoidal — APPROVED** [review]
LGTM!
@victormartinelicocorp @seb-elico Please review

File: business_requirement_deliverable_resource_template/__openerp__.py
Comment (elicoidal) [inline business_requirement_deliverable_resource_template/__openerp__.py]: should be _demo

File: business_requirement_deliverable_resource_template_categ/__openerp__.py
Comment (elicoidal) [inline business_requirement_deliverable_resource_template_categ/__openerp__.py]: should be _demo

File: business_requirement_deliverable_cost/models/business.py
Comment (martintorresvm) [inline business_requirement_deliverable_cost/models/business.py:125]: @darshan-serpent This should be part of another PR.

Seems not too big changes and makes sense so at least I would like know if it's documented this Issue or what is about. Could you add some reference?

Comment (elicoidal) [reply business_requirement_deliverable_cost/models/business.py:125]: @victormartinelicocorp This is documented in the conversation but not in a formal issue.
I would keep it here for the sake of simplicity

Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py:125]: @elicoidal Was just a nice to have, the others LGTM

**Review by martintorresvm — APPROVED** [review]

File: business_requirement_deliverable_resource_template_categ/data/business_requirement_deliverable_resource_template_categ_demo.xml
Comment (seb-elico) [inline business_requirement_deliverable_resource_template_categ/data/business_requirement_deliverable_resource_template_categ_demo.xml]: Could you please fix the indentation of this line? `field` is child of `record` (same for all the document)

**Review by seb-elico — CHANGES_REQUESTED** [review]

**Comment by seb-elico** [comment]
@darshan-serpent @victormartinelicocorp any reason why the coverage has decreased?

**Comment by martintorresvm** [comment]
> @darshan-serpent @victormartinelicocorp any reason why the coverage has decreased?

@seb-elico Modify functions not covered:
https://github.com/OCA/business-requirement/pull/212/files#diff-63f164bb06a74e20b85ab6efc5bb42ffR125

Coverage still quite good but sure @darshan-serpent could fix it.

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp @seb-elico

**Review by seb-elico — APPROVED** [review]
LGTM

**Pull request merged**
